tests/zkasm/helpers/InstructionTracer.js

```diff
@@ -0,0 +1,47 @@
+const fs = require("fs");
+
+/**
+ * Handles the generation of traces of instructions executed at runtime.
+ */
+class InstructionTracer {
+    constructor() {
+        // Contains executed instructions in order of execution.
+        this.rawTrace = [];
+    }
+
+    setup() {
+        // `zkevm-proverjs` requires a `setup` function on helper objects.
+    }
+
+    /**
+     * @param {Object} ctx - The context passed by `zkevm-proverjs` to helper calls
+     * @param {Object} tag - Helper call specific information passed by `zkevm-proverjs`
+     * @param {Object} tag.params[0] - expects the identifier as first parameter of
+     *   `$${instrumentInst(...)}`
+     * 
+     * @example
+     * // To trace the execution of an `AluRRR` instruction, call
+     * // $${instrumentInst(AluRRR)}
+     */
+    eval_traceInstruction(ctx, tag) {
+        const instruction = tag.params[0].varName;
+        this.rawTrace.push(instruction);
+    }
+
+    /**
+     * Writes the raw trace to `path`.
+     * 
+     * @param {string} path 
+     */
+    writeRawTrace(path) {
+        if (typeof path !== "string") {
+            // Writing to a descriptor will append instead of replace content,
+            // which might result in invalid traces, see 
+            // https://nodejs.org/api/fs.html#using-fswritefile-with-file-descriptors
+            throw new Error("provide a file name (not descriptor) to write to");
+        }
+        fs.writeFileSync(path, this.rawTrace.join("\n"));
```

Comment:
How large do these traces get for real programs we want to look at? My intuition is that they are going to become pretty large after executing for not too long, so this code may very well be pretty limited by memory available on the machine. An easy improvement here would be to join and write out the instructions in chunks of, say, 1000 instructions or somesuch number. That would likely cut down the peak memory use by this in half with a reasonably little effort.

(Another future improvement that's more involved would be to collect into `this.rawTrace` numeric IDs of instructions into a `UInt8Array`/`Buffer` and write them out as a binary blob, that way a trace of 1M instructions would end up being just 1MB, and not a significant multiple of the number.)

Reply:
Though now that I think about it, we've a limit of 8M steps, don't we? So the trace is still in the realm of megabytes, which is most likely perfectly fine.